cmake/FindPylon.cmake

```diff
@@ -0,0 +1,159 @@
+#############################################################################
```

Comment:
I'll try to figure out if adding this file affects the license of our other files.

Reply:
I just found [this one](https://github.com/basler/pylon-ros-camera/blob/master/pylon_camera/cmake/FindPylon.cmake), can you give it a try?

It is part of the ROS package provided by Basler, so I would assume it should work.  The package is licensed with BSD, so we can use this without headache.

In case you use this file, please add a comment header with a link to the original file and simply put the complete content of [this LICENSE file](https://github.com/basler/pylon-ros-camera/blob/master/pylon_camera/LICENSE) there as well.

Reply:
This cmake file's not working. I will look into the variable names, but ideally this should just work, right?

Reply:
Ok I changed all "Pylon"s to "PYLON"s and it works now

Reply:
I think it would be better to do it the other way round, i.e. keep the FindPylon.cmake as it is and change to `Pylon_` in our CMakeLists.txt.  This way it would be more compliant with CMake conventions.

Reply:
I changed it PYLON in the find_package then, that's fine right?

Reply:
Do we need to follow the CMake convention? I was just thinking it's cleaner now since it's in caps (as PYLON) everywhere except in find_package() where it's "Pylon". Otherwise it's a mix of both everywhere.

Reply:
Yes, we should follow these conventions. It makes the code much easier to use because you don't need to dig and figure out how the variables are named.

What I meant is to change it to `Pylon_` everywhere, so there would be no mix.